smart_open/smart_open_lib.py

```diff
@@ -157,20 +168,26 @@ def smart_open(uri, mode="rb", **kw):
         elif parsed_uri.scheme in ("s3", "s3n", 's3u'):
             return s3_open_uri(parsed_uri, mode, **kw)
         elif parsed_uri.scheme in ("hdfs", ):
+            if kw.get('encoding') is not None:
+                warnings.warn('Issue #146: encoding is not supported for HDFS, ignoring')
```

Comment:
Let's be a little more verbose. Why is not supported -- is it a fault of the user, or a functionality gap in `smart_open`? What can the user do about this? (how to fix the warning?)

Reply:
OK, done. Please have a look.